### ISSUE TITLE
collector "bgpl2vpn" scrape failed: cannot unmarshal outputs of 'show evpn vni json'

### ISSUE THREAD

**Issue by matejv** [tynany/frr_exporter#22]
I'm running frr_exporter 0.2.11 on Cumulus Linux 4.2.0. 

This is the command to run the exporter:

```
frr_exporter --collector.bgp6 --collector.bgpl2vpn --no-collector.ospf --web.listen-address=:9199
```

On every poll I'm seeing this error in the logs:

```
ERRO[0003] collector "bgpl2vpn" scrape failed: cannot unmarshal outputs of 'show evpn vni json': json: cannot unmarshal string into Go struct field vxLanStats.NumRemoteVteps of type float64  source="collector.go:138"
```

And the scrape for `bgpl2vpn` fails:

```
frr_collector_up{collector="bgpl2vpn"} 0
frr_scrape_errors_total{collector="bgpl2vpn"} 24255
```

If I run the show command by hand I get this output:

```json
myswitch# show evpn vni json
{
  "101104":{
    "vni":101104,
    "type":"L2",
    "vxlanIf":"vxlan1104",
    "numMacs":7,
    "numArpNd":0,
    "numRemoteVteps":2,
    "tenantVrf":"default",
    "remoteVteps":[
      "10.99.1.224",
      "10.99.1.121"
    ]
  },
  "101195":{
    "vni":101195,
    "type":"L2",
    "vxlanIf":"vxlan1195",
    "numMacs":3,
    "numArpNd":0,
    "numRemoteVteps":1,
    "tenantVrf":"default",
    "remoteVteps":[
      "10.99.1.224"
    ]
  },
  "100042":{
    "vni":100042,
    "type":"L2",
    "vxlanIf":"vxlan42",
    "numMacs":4,
    "numArpNd":0,
    "numRemoteVteps":2,
    "tenantVrf":"default",
    "remoteVteps":[
      "10.99.1.224",
      "10.99.1.121"
    ]
  },
  "101108":{
    "vni":101108,
    "type":"L2",
    "vxlanIf":"vxlan1108",
    "numMacs":0,
    "numArpNd":0,
    "numRemoteVteps":2,
    "tenantVrf":"default",
    "remoteVteps":[
      "10.99.1.224",
      "10.99.1.121"
    ]
  },
  "101103":{
    "vni":101103,
    "type":"L2",
    "vxlanIf":"vxlan1103",
    "numMacs":15,
    "numArpNd":0,
    "numRemoteVteps":2,
    "tenantVrf":"default",
    "remoteVteps":[
      "10.99.1.224",
      "10.99.1.121"
    ]
  },
  "100041":{
    "vni":100041,
    "type":"L2",
    "vxlanIf":"vxlan41",
    "numMacs":6,
    "numArpNd":0,
    "numRemoteVteps":2,
    "tenantVrf":"default",
    "remoteVteps":[
      "10.99.1.224",
      "10.99.1.121"
    ]
  },
  "101115":{
    "vni":101115,
    "type":"L2",
    "vxlanIf":"vxlan1115",
    "numMacs":12,
    "numArpNd":0,
    "numRemoteVteps":2,
    "tenantVrf":"default",
    "remoteVteps":[
      "10.99.1.224",
      "10.99.1.121"
    ]
  },
  "101109":{
    "vni":101109,
    "type":"L2",
    "vxlanIf":"vxlan1109",
    "numMacs":0,
    "numArpNd":0,
    "numRemoteVteps":2,
    "tenantVrf":"default",
    "remoteVteps":[
      "10.99.1.224",
      "10.99.1.121"
    ]
  },
  "101106":{
    "vni":101106,
    "type":"L2",
    "vxlanIf":"vxlan1106",
    "numMacs":12,
    "numArpNd":0,
    "numRemoteVteps":2,
    "tenantVrf":"default",
    "remoteVteps":[
      "10.99.1.224",
      "10.99.1.121"
    ]
  },
  "101100":{
    "vni":101100,
    "type":"L2",
    "vxlanIf":"vxlan1100",
    "numMacs":11,
    "numArpNd":0,
    "numRemoteVteps":2,
    "tenantVrf":"default",
    "remoteVteps":[
      "10.99.1.224",
      "10.99.1.121"
    ]
  },
  "100049":{
    "vni":100049,
    "type":"L2",
    "vxlanIf":"vxlan49",
    "numMacs":0,
    "numArpNd":0,
    "numRemoteVteps":2,
    "tenantVrf":"default",
    "remoteVteps":[
      "10.99.1.224",
      "10.99.1.121"
    ]
  },
  "100045":{
    "vni":100045,
    "type":"L2",
    "vxlanIf":"vxlan45",
    "numMacs":0,
    "numArpNd":0,
    "numRemoteVteps":2,
    "tenantVrf":"default",
    "remoteVteps":[
      "10.99.1.224",
      "10.99.1.121"
    ]
  },
  "104005":{
    "vni":104005,
    "vxlanIf":"vxlan4005",
    "numMacs":2,
    "numArpNd":3,
    "numRemoteVteps":"n\/a",
    "type":"L3",
    "tenantVrf":"fwlan-2"
  },
  "104004":{
    "vni":104004,
    "vxlanIf":"vxlan4004",
    "numMacs":1,
    "numArpNd":1,
    "numRemoteVteps":"n\/a",
    "type":"L3",
    "tenantVrf":"fwlan-1"
  },
  "104001":{
    "vni":104001,
    "vxlanIf":"vxlan4001",
    "numMacs":4,
    "numArpNd":9,
    "numRemoteVteps":"n\/a",
    "type":"L3",
    "tenantVrf":"internet"
  },
  "104002":{
    "vni":104002,
    "vxlanIf":"vxlan4002",
    "numMacs":1,
    "numArpNd":2,
    "numRemoteVteps":"n\/a",
    "type":"L3",
    "tenantVrf":"internet-2"
  }
}
```

I suspect the `n/a` values are problematic.

**Comment by eliasrudberg** [comment]
I encountered a similar problem, started after the commits a few days ago:

    msg="collector \"bfd\" scrape failed: cannot unmarshal bfd peers json: unexpected end of JSON input" source="collector.go:138"

**Comment by tynany** [comment]
@matejv thanks for the detailed bug report. I have fixed this in the latest release, 0.2.12 (https://github.com/tynany/frr_exporter/releases/tag/v0.2.12).

@eliasrudberg, you are receiving this error as the latest version supports BFD. If you do not use BFD, you can disable the collector by passing the `--no-collector.bfd` flag to `frr_exporter`.

**Comment by matejv** [comment]
@tynany Yes, it works now! No more errors and I get l2vpn metrics:

```
# HELP frr_bgp_l2vpn_evpn_remote_vtep_count_total Number of known remote VTEPs. A value of -1 indicates a non-integer output from FRR, such as n/a.
# TYPE frr_bgp_l2vpn_evpn_remote_vtep_count_total gauge
frr_bgp_l2vpn_evpn_remote_vtep_count_total{tenantVrf="fwlan-1",type="L3",vni="104004",vxlanIf="vxlan4004"} -1
frr_bgp_l2vpn_evpn_remote_vtep_count_total{tenantVrf="fwlan-2",type="L3",vni="104005",vxlanIf="vxlan4005"} -1
frr_bgp_l2vpn_evpn_remote_vtep_count_total{tenantVrf="internet",type="L2",vni="101119",vxlanIf="vxlan1119"} 1
frr_bgp_l2vpn_evpn_remote_vtep_count_total{tenantVrf="internet",type="L3",vni="104001",vxlanIf="vxlan4001"} -1
frr_bgp_l2vpn_evpn_remote_vtep_count_total{tenantVrf="internet-2",type="L3",vni="104002",vxlanIf="vxlan4002"} -1
```
